src/cmdstan/command_helper.hpp

```diff
@@ -635,6 +672,158 @@ void services_log_prob_grad(const stan::model::model_base &model, bool jacobian,
   }
 }
 
+/**
+ * For sample and pathfinder methods, return number of
+ * chains or pathfinders to run, otherwise return 1.
+ * For the sampler, only adaptive NUTS-HMC with non-unit metric
+ * supports multi-chain parallelization; if sampler config is invalid,
+ * this will throw an error.
+ *
+ * @param parser user config
+ * @return int num chains or paths
+ */
+unsigned int get_num_chains(argument_parser &parser) {
+  auto user_method = parser.arg("method");
+  if (user_method->arg("pathfinder"))
+    return get_arg_val<int_argument>(parser, "method", "pathfinder",
+                                     "num_paths");
+  if (!user_method->arg("sample"))
+    return 1;
+  unsigned int num_chains
+      = get_arg_val<int_argument>(parser, "method", "sample", "num_chains");
+  if (num_chains > 1)
+    validate_multi_chain_config(user_method);
+  return num_chains;
+}
+
+/**
+ * Check possible name conflicts between input and output files where both
+ * are in Stan CSV format; if conflicts detected, quit with an error message.
+ *
+ * @param parser user config
+ */
+void check_file_config(argument_parser &parser) {
+  std::string sample_file
+      = get_arg_val<string_argument>(parser, "output", "file");
+  std::string input_file;
+  auto user_method = parser.arg("method");
+  if (user_method->arg("generate_quantities")) {
+    input_file = get_arg_val<string_argument>(
+        parser, "method", "generate_quantities", "fitted_params");
+    if (input_file.empty()) {
+      throw std::invalid_argument(
+          std::string("Argument fitted_params file - found empty string, "
+                      "expecting filename."));
+      if (input_file.compare(sample_file) == 0) {
+        std::stringstream msg;
+        msg << "Filename conflict, fitted_params file " << input_file
+            << " and output file names are identical, must be different."
+            << std::endl;
+        throw std::invalid_argument(msg.str());
```

Comment:
Can you just use an std string here?

Reply:
I think the filename is useful information - is that what you're asking?  or are you suggesting avoid setting up the std::stringstream and just constructing the message as argument to the `std::invalid_argument` constructor?

I would prefer to leave it as is because
+  this makes it easy to see the message
+ it's consistent with error messaging elsewhere.